src/BSON/Timestamp.c

```diff
@@ -226,6 +232,102 @@ PHP_METHOD(Timestamp, __wakeup)
 }
 /* }}} */
 
+/* {{{ proto string Timestamp::serialize()
+*/
+PHP_METHOD(Timestamp, serialize)
+{
+	php_phongo_timestamp_t   *intern;
+#if PHP_VERSION_ID >= 70000
+	zval                      retval;
+#else
+	zval                     *retval;
+#endif
+	php_serialize_data_t      var_hash;
+	smart_str                 buf = { 0 };
+	char                      s_increment[24];
+	char                      s_timestamp[24];
+	int                       s_increment_len;
+	int                       s_timestamp_len;
+
+	intern = Z_TIMESTAMP_OBJ_P(getThis());
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	s_increment_len = snprintf(s_increment, sizeof(s_increment), "%" PRId64, intern->increment);
+	s_timestamp_len = snprintf(s_timestamp, sizeof(s_timestamp), "%" PRId64, intern->timestamp);
```

Comment:
On second glance, `php_phongo_timestamp_init_from_string()` is using `int64_t` due to the string conversion functions. It relies on `php_phongo_timestamp_init()`, which does the range checking.

Perhaps `php_phongo_timestamp_init()` has an issue on 32-bit platforms, since its arguments are `phongo_long` types, which would be a signed 32-bit value on some platforms. In any event, that's clearly outside the scope of this PR.

My first comment in this thread still stands, though.